extension/storage/dbstorage/config.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dbstorage // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage/dbstorage"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+// Config defines configuration for dbstorage extension.
+type Config struct {
+	config.ExtensionSettings `mapstructure:",squash"`
+	DriverName               string `mapstructure:"driver,omitempty"`
+	DataSource               string `mapstructure:"datasource,omitempty"`
+}
+
+func (cfg *Config) Validate() error {
+	if cfg.DataSource == "" {
+		return fmt.Errorf(fmt.Sprintf("missing datasource for %s", cfg.ID()))
+	}
+	if cfg.DriverName == "" {
```

Comment:
Should it accept only the two currently known values?

Reply:
I think downstream distributions could add additional drivers. I just added sqlite3 and postgres for now, but it's easy to add more later as well here.